src/utils.ts

```diff
@@ -79,37 +84,148 @@ export const fetchTransactionLogs = async (
   ).flat();
 };
 
+export const convertJson = (previousData: any) => {
+  const updatedData: Record<string, IPositions> = {};
+
+  for (const userId in previousData) {
+    const userPools = previousData[userId];
+
+    const updatedActive = userPools.active.map((activeEntry: any) => {
+      const updatedEvent = {
+        ...activeEntry.event,
+        id: new BN(activeEntry.event.id, "hex"),
+        liquidity: new BN(activeEntry.event.liquidity, "hex"),
+        currentTimestamp: new BN(activeEntry.event.currentTimestamp, "hex"),
+      };
+      return { event: updatedEvent };
+    });
+
+    const updatedClosed = userPools.closed.map((closedEntry: any) => {
+      const openEvent = !closedEntry.events[0]
+        ? closedEntry.events[0]
+        : {
+            ...closedEntry.events[0],
+            id: new BN(closedEntry.events[0].id, "hex"),
+            liquidity: new BN(closedEntry.events[0].liquidity, "hex"),
+            currentTimestamp: new BN(
+              closedEntry.events[0].currentTimestamp,
+              "hex"
+            ),
+          };
+      const closeEvent = {
+        ...closedEntry.events[1],
+        id: new BN(closedEntry.events[1].id, "hex"),
+        liquidity: new BN(closedEntry.events[1].liquidity, "hex"),
+        currentTimestamp: new BN(closedEntry.events[1].currentTimestamp, "hex"),
+        upperTickSecondsPerLiquidityOutside: new BN(
+          closedEntry.events[1].upperTickSecondsPerLiquidityOutside,
+          "hex"
+        ),
+        lowerTickSecondsPerLiquidityOutside: new BN(
+          closedEntry.events[1].lowerTickSecondsPerLiquidityOutside,
+          "hex"
+        ),
+        poolSecondsPerLiquidityGlobal: new BN(
+          closedEntry.events[1].poolSecondsPerLiquidityGlobal,
+          "hex"
+        ),
+      };
+      const updatedEvents = [openEvent, closeEvent];
+      return { events: updatedEvents };
+    });
+
+    updatedData[userId] = {
+      active: updatedActive,
+      closed: updatedClosed,
+    };
+  }
+  return updatedData;
+};
+
 export const extractEvents = (
-  initialEvents: any,
+  previousData: Record<string, IPositions>,
   market: Market,
   transactionLog: string[]
-) => {
-  const eventsObject = initialEvents;
-  const eventLogs = transactionLog.filter((log) =>
-    log.startsWith("Program data:")
-  );
-  eventLogs.forEach((eventLog) => {
-    const decodedEvent = market.eventDecoder.decode(
-      eventLog.split("Program data: ")[1]
+): Record<string, IPositions> => {
+  const eventsObject: Record<string, IPositions> = { ...previousData };
+
+  const eventLogs = transactionLog
+    .filter((log) => log.startsWith("Program data:"))
+    .map((log) => log.split("Program data: ")[1]);
+
+  const isPromotedPool = (pool: any) =>
```

Comment:
It will be better to avoid using `any` type. It should be `PublicKey`
Is it necessary to nest the funcitons declaration inside parent one?